mojap_metadata/metadata/metadata.py

```diff
@@ -0,0 +1,115 @@
+import json
+from warnings import warn
+
+import jsonschema
+
+
+class MetadataProperty:
+    def __set_name__(self, owner, name) -> None:
+        self.name = f"_{name}"
+
+    def __get__(self, obj, type=None) -> object:
+        return getattr(obj, self.name)
+
+    def __set__(self, obj, value) -> None:
+        setattr(obj, self.name, value)
+        obj.validate()
+
+
+class Metadata:
+    name = MetadataProperty()
+    description = MetadataProperty()
+    file_format = MetadataProperty()
+    sensitive = MetadataProperty()
+    columns = MetadataProperty()
+    primary_key = MetadataProperty()
+    partitions = MetadataProperty()
+
+    def __init__(
+        self,
+        name: str = "",
+        description: str = "",
+        file_format: str = "",
+        sensitive: bool = False,
+        columns: list = [],
+        primary_key: set = [],
+        partitions: set = [],
+    ) -> None:
+        self._name = name
+        self._description = description
+        self._file_format = file_format
+        self._sensitive = sensitive
+        self._columns = columns
+        self._primary_key = primary_key
+        self._partitions = partitions
+
+        with open("mojap_metadata/metadata/specs/table_schema.json") as file:
+            self._schema = json.load(file)
```

Comment:
I think this might be better off being attributed to the class? So that it is only loaded once? Also worth thinking about pointing that to a web address in the future so we can always pull the latest version the schema? This latter point might be worth adding as an issue.

Reply:
created new issue #21